api/base/exceptions.py

```diff
@@ -80,20 +80,82 @@ class Conflict(APIException):
     default_detail = 'Query string contains an invalid value.'
     status_code = http.BAD_REQUEST
 
-    def __init__(self, detail=None, parameter=None):
-        super(InvalidQueryStringError, self).__init__(detail=detail, source={
-            'parameter': parameter
-        })
 
+class InvalidFilterOperator(JSONAPIParameterException):
+    """Raised when client passes an invalid operator to a query param filter."""
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None):
+        if value and not detail:
+            detail = "Value '{0}' is not a supported filter operator; use one of eq, lt, lte, gt, gte.".format(value)
+        super(InvalidFilterOperator, self).__init__(detail=detail, parameter='filter')
+
+
+class InvalidFilterValue(JSONAPIParameterException):
+    """Raised when client passes an invalid value to a query param filter."""
+    default_detail = 'Invalid filter value'
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None, field_type=None):
+        if value and not detail:
+            detail = "Value '{0}' is not valid for a filter on type {1}.".format(
+                value,
+                field_type
+            )
+        super(InvalidFilterValue, self).__init__(detail=detail, parameter='filter')
+
+
+class InvalidFilterError(JSONAPIParameterException):
+    """Raised when client passes an malformed filter in the query string."""
+    default_detail = 'Query string contains a malformed filter.'
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None):
+        super(InvalidFilterError, self).__init__(detail=detail, parameter='filter')
+
+
+class InvalidFilterComparisonType(JSONAPIAttributeException):
+    """Raised when client tries to filter on a field that is not a date or number type"""
+    default_detail = "Comparison operators are only supported for dates and numbers."
+    status_code = http.BAD_REQUEST
+
+
+class InvalidFilterMatchType(JSONAPIAttributeException):
+    """Raised when client tries to do a match filter on a field that is not a string or a list"""
+    default_detail = "Match operators are only supported for strings and lists."
+    status_code = http.BAD_REQUEST
+
+
+class InvalidFilterFieldError(JSONAPIAttributeException):
+    """Raised when client tries to filter on a field that is not supported"""
+    default_detail = "Query contained one or more filters for invalid fields."
+    status_code = http.BAD_REQUEST
 
-class InvalidFilterError(ParseError):
-    """Raised when client passes an invalid filter in the query string."""
-    default_detail = 'Query string contains an invalid filter.'
+    def __init__(self, detail=None, parameter=None, value=None):
+        if value and not detail:
+            detail = "Value '{}' is not a filterable field.".format(value)
+        super(InvalidFilterFieldError, self).__init__(detail=detail, parameter=parameter)
```

Comment:
Does this work as expected? You are passing `parameter`, but `parameter` isn't a parameter of the superclass.